src/charm.py

```diff
@@ -93,6 +94,11 @@ def __init__(self, *args):
             self.framework.observe(self.on[rel].relation_changed, self._on_event)
 
         self._logging = LogForwarder(charm=self)
+        self._scraping = MetricsEndpointProvider(
```

Comment:
Should we keep this consistent among charms? I just noticed that previously we were using[prometheus_provider](https://github.com/canonical/istio-operators/pull/477/files#diff-d4a88abbd5dcb0911fc960c33c9f0440bff32b2ddc60d56d266788300b709e02R38).